src/validateChangelog.js

```diff
@@ -1,112 +0,0 @@
-const { parseChangelog } = require('./parseChangelog');
-
-/**
- * @typedef {import('./constants.js').Version} Version
- */
-
-/**
- * Indicates that the changelog is invalid.
- */
-class InvalidChangelogError extends Error {}
-
-/**
- * Indicates that unreleased changes are still present in the changelog.
- */
-class UnreleasedChangesError extends InvalidChangelogError {
-  constructor() {
-    super('Unreleased changes present in the changelog');
-  }
-}
-
-/**
- * Indicates that the release header for the current version is missing.
- */
-class MissingCurrentVersionError extends InvalidChangelogError {
-  /**
-   * @param {Version} currentVersion - The current version
-   */
-  constructor(currentVersion) {
-    super(`Current version missing from changelog: '${currentVersion}'`);
-  }
-}
-
-/**
- * Represents a formatting error in a changelog.
- */
-class ChangelogFormattingError extends InvalidChangelogError {
-  /**
-   * @param {Object} options
-   * @param {string} options.validChangelog - The string contents of the well-
-   *   formatted changelog.
-   * @param {string} options.invalidChangelog - The string contents of the
-   *   malformed changelog.
-   */
-  constructor({ validChangelog, invalidChangelog }) {
-    super('Changelog is not well-formatted');
-    this.data = {
-      validChangelog,
-      invalidChangelog,
-    };
-  }
-}
-
-/**
- * Validates that a changelog is well-formatted.
- * @param {Object} options
- * @param {string} options.changelogContent - The current changelog
- * @param {Version} options.currentVersion - The current version
- * @param {string} options.repoUrl - The GitHub repository URL for the current
- *   project.
- * @param {boolean} options.isReleaseCandidate - Denotes whether the current
- *   project is in the midst of release preparation or not. If this is set, this
- *   command will also ensure the current version is represented in the
- *   changelog with a release header, and that there are no unreleased changes
- *   present.
- * @throws {InvalidChangelogError} Will throw if the changelog is invalid
```

Comment:
Despite the lack of VSCode support, I do think we should preserve these as documentation. They're pretty important for knowing how to properly use the function. As CLI.js does for example.

Reply:
Restored in 78402b7